### PR TITLE
Replace vega-utils.replace() with spread operator

### PR DESCRIPTION
**Pull request by cjiang**
This is the refactoring proposed in #1953.

The merge() in src/compile/data/bin.ts - is that intended?

### CODE REVIEW

**Comment by domoritz** [comment]
> The merge() in src/compile/data/bin.ts - is that intended?

We use the merge method to combine nodes in the dataflow. Right now we don't optimize the dataflow very much but in the future, we will need these functions more.

File: src/compile/selection/selection.ts [hunk 1]
```diff
@@ -90,10 +90,10 @@ export function parseUnitSelection(model: UnitModel, selDefs: Dict<SelectionDef>
     }
 
     name = varName(name);
-    const selCmpt = selCmpts[name] = extend({}, selDef, {
+    const selCmpt = selCmpts[name] = {...selDef,
```
Comment (kanitw) [inline src/compile/selection/selection.ts]: I would add a line break between `{` and `...`

**Comment by cjiang** [comment]
To clarify, I mean in bins.ts, merge() is implemented as:
`this.bins = {...other.bins};`
which suspiciously doesn't involve this.bins.

File: src/compile/selection/selection.ts (hunk 1 above)
Comment (cjiang) [reply src/compile/selection/selection.ts]: Done.

**Comment by domoritz** [comment]
> this.bins = {...other.bins};

Good catch @cjiang! Yes, that should be `this.bins = {...this.bins, ...other.bins};`!

**Comment by kanitw** [comment]
Thanks for the PR :)

**Pull request merged**